File: backend/api/schemas/salary.py
"""
api/schemas/salary.py
=====================
Pydantic response schemas for the Salary Intelligence router.
"""

from pydantic import BaseModel


class RoleSalary(BaseModel):
    """Average salary figures for a specific job role."""
    role_name: str
    role_category: str
    seniority_level: str
    avg_salary_usd: float
    data_points: int            # number of role×country×year records


class CountrySalary(BaseModel):
    """Average salary figures for a specific country."""
    country_name: str
    country_code: str
    region: str
    avg_salary_usd: float
    data_points: int


class SalaryTrendPoint(BaseModel):
    """One point on a salary-over-time trend chart."""
    year: int
    avg_salary_usd: float
    salary_growth_pct: float | None   # NULL for the first year (no baseline)
    median_salary_usd: float | None


class TopPayingRole(BaseModel):
    """A role in the top-N highest-paying list."""
    rank: int
    role_name: str
    role_category: str
    seniority_level: str
    avg_salary_usd: float


class TopPayingCountry(BaseModel):
    """A country in the top-N highest-paying list."""
    rank: int
    country_name: str
    country_code: str
    region: str
    avg_salary_usd: float

File: frontend/src/types/salary.ts
// ─────────────────────────────────────────
// types/salary.ts
// Field names mirror backend/api/schemas/salary.py exactly.
// ─────────────────────────────────────────

// GET /api/v1/salary/by-role
export interface RoleSalary {
  role_name: string
  role_category: string
  seniority_level: string
  avg_salary_usd: number
  data_points: number
}

// GET /api/v1/salary/by-country
export interface CountrySalary {
  country_name: string
  country_code: string
  region: string
  avg_salary_usd: number
  data_points: number
}

// GET /api/v1/salary/trends
export interface SalaryTrendPoint {
  year: number
  avg_salary_usd: number
  salary_growth_pct: number | null    // null for first year (no baseline)
  median_salary_usd: number | null
}

// GET /api/v1/salary/top-paying-roles
export interface TopPayingRole {
  rank: number
  role_name: string
  role_category: string
  seniority_level: string
  avg_salary_usd: number
}

// GET /api/v1/salary/top-paying-countries
export interface TopPayingCountry {
  rank: number
  country_name: string
  country_code: string
  region: string
  avg_salary_usd: number
}

// Aggregate type for the page hook
export interface SalaryData {
  byRole: RoleSalary[]
  byCountry: CountrySalary[]
  trends: SalaryTrendPoint[]
  topRoles: TopPayingRole[]
  topCountries: TopPayingCountry[]
}